src/components/AddPayment.vue

```diff
@@ -0,0 +1,227 @@
+            >※1文字以上20文字以内でご記入ください</span
+          >
+        </div>
+
+        <div class="input-price-form">
+          <p>支払い金額</p>
+          <div class="price-box">
+            <input
+              type="text"
+              id="price"
+              placeholder="例：1000"
+              v-model="inputAmount"
+            />
+            <div class="money-unit">
+              <span>{{ moneyUnit }}</span>
+            </div>
+          </div>
+          <span v-bind:class="{ red: inputAmountError }">{{
+            inputAmountErrorText
+          }}</span>
+        </div>
+
+        <div class="pulldown-payer-form">
+          <p>支払ったメンバー</p>
+          <div class="pulldown-box">
+            <select v-model="payer_id">
+              <option
+                v-for="member in members"
+                :key="member.id"
+                v-bind:value="member.id"
+              >
+                {{ member.name }}
+              </option>
+            </select>
+          </div>
+        </div>
+
+        <div id="select-borrower-wrapper">
+          <p>誰の分を払ったか</p>
+          <div
+            v-for="(member, index) in members"
+            :key="member.id"
+            class="select-borrower-box"
+          >
+            <p>メンバー{{ index + 1 }}</p>
+            <div class="member-name-box">
+              <input
+                type="checkbox"
+                :id="'member' + member.id"
+                :value="member.id"
+                v-model="borrowers"
+              />
+              <label :for="'member' + member.id">
+                <span>{{ member.name }}</span>
+              </label>
+            </div>
+          </div>
+        </div>
+
+        <div class="button-wrapper">
+          <div class="add-payment-button-wrapper">
+            <button class="add-payment-button" @click="doValidation">
+              <span>追加</span>
+            </button>
+          </div>
+          <div class="back-button-wrapper">
+            <button class="back-button" @click="toGroup">
+              <span>キャンセル</span>
+            </button>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { api_request } from "../js/api.js";
+import { SEISANKUN_API_BASE_URL } from "../javascripts/config.js";
+export default {
+  data() {
+    return {
+      members: [],
+      inputPaymentTitle: "",
+      inputAmount: "",
+      payer_id: null,
+      inputPaymentTitleError: false,
+      inputAmountError: false,
+      travel_key: "",
+      inputAmountErrorText: "※半角数字でご記入ください",
+      moneyUnit: "円",
+      borrowers: [],
+    };
+  },
+  methods: {
+    doValidation() {
+      let errors = 0;
+      //支払い内容のバリデーション
+      if (
+        this.inputPaymentTitle.trim().length >= 1 &&
+        this.inputPaymentTitle.trim().length <= 20
+      ) {
+        this.inputPaymentTitleError = false;
+      } else {
+        this.inputPaymentTitleError = true;
+        errors++;
+      }
```

Comment:
[should]
初期値は決まっているので、わざわざfalseを入れる必要はないと思った。
以下のようにしていいと思う

```suggestion
      if (
        this.inputPaymentTitle.trim().length < 1 &&
        this.inputPaymentTitle.trim().length > 20
      ) {
        this.inputPaymentTitleError = true;
        errors++;
      }
```

Reply:
この記述には理由があって、

支払い内容と支払い金額の両方が空の状態で「追加」を押すと、義広の提案するコードでも支払い内容のinputPaymentTitleErrorがtrueになってエラー文が表示されるまでは良いんだけど、

次に、支払い内容が入力済み、支払い金額が空で「追加」を押すと、支払い内容のエラー文が消えないから、使い勝手が悪くなると思って、falseを設定できるようにもしてあるので、これはこのままでいこうと思ってる。